File: app/src/main/java/seveida/firetvforreddit/response/objects/SubredditResponse.java
package seveida.firetvforreddit.response.objects;

import com.squareup.moshi.Json;

public class SubredditResponse {

    @Json(name = "kind")
    public String kind;
    @Json(name = "data")
    public Data data;

    /**
     * No args constructor for use in serialization
     *
     */
    public SubredditResponse() {
    }

    /**
     *
     * @param data
     * @param kind
     */
    public SubredditResponse(String kind, Data data) {
        super();
        this.kind = kind;
        this.data = data;
    }

}

File: app/src/main/java/seveida/firetvforreddit/response/objects/SubredditResponse.kt
package seveida.firetvforreddit.response.objects

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SubredditResponse (
    @Json(name = "kind") val kind: String,
    @Json(name = "data") val data: Data
)
